backend/src/main/java/org/uresti/pozarreal/service/impl/FilesStorageServiceImpl.java

```diff
@@ -63,6 +82,41 @@ public Path save(MultipartFile file) {
         }
     }
 
+    @Override
+    public String save(MultipartFile file, LoggedUser loggedUser, String paymentId) {
+
+        paymentRepository.findById(paymentId).orElseThrow();
+
+        DbxRequestConfig config = DbxRequestConfig.newBuilder(dropBoxConfig.getClientIdentifier()).build();
+        DbxClientV2 client = new DbxClientV2(config, dropBoxConfig.getAccessToken());
+
+        try {
+            String originalName = file.getOriginalFilename();
+            assert originalName != null;
+            String extension = originalName.substring(originalName.indexOf('.'));
+            String fileName = UUID.randomUUID().toString();
+            client.files()
+                    .uploadBuilder("/" + fileName + extension)
+                    .withMode(WriteMode.ADD)
+                    .uploadAndFinish(file.getInputStream());
+
+            SharedLinkMetadata sharingRequests = client.sharing()
+                    .createSharedLinkWithSettings("/" + fileName + extension);
+
+            PaymentEvidence paymentEvidence = new PaymentEvidence();
+
+            paymentEvidence.setId(UUID.randomUUID().toString());
+            paymentEvidence.setPaymentId(paymentId);
+            paymentEvidence.setUrl(sharingRequests.getUrl());
+
+            paymentEvidenceRepository.save(paymentEvidence);
+
+            return sharingRequests.getUrl();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not store the file. Error: " + e.getMessage());
```

Comment:
throw instead a PozarrealSystemException.. you can see an example in DBUpdaterServiceImpl... When you catch a exception and throw other instead remember always to nest the exceptions, so the stack trace can contain the root cause always